### PR TITLE
Add Proxycurl client

### PR DESCRIPTION
**Pull request by collindutter**

### CODE REVIEW

File: griptape/labs/tools/proxycurl/tool.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+from griptape.core import BaseTool
+from griptape.core.decorators import activity
+from schema import Schema, Literal
+from attr import define, field
+import requests
+
+
+@define
+class ProxycurlClient(BaseTool):
+    api_key: str = field(default=None, kw_only=True)
```
Comment (vasinov) [inline griptape/labs/tools/proxycurl/tool.py]: Should be either `Optional[str]`, or no default `None` value, or a default empty string value.

Seems like making it optional makes sense and then `header_dic` could be conditional.

File: griptape/labs/tools/proxycurl/tool.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        profile_id = params["values"]["profile_id"]
+        header_dic = {"Authorization": "Bearer " + self.api_key}
+        params = {
+            "url": f"https://www.linkedin.com/in/{profile_id}",
+        }
+
+        response = requests.get(
+            self.api_endpoint, params=params, headers=header_dic, timeout=30
+        )
+        print(response.json())
```
Comment (vasinov) [inline griptape/labs/tools/proxycurl/tool.py]: Cleanup print.

File: griptape/labs/tools/proxycurl/tool.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    )
+    def get_profile(self, params: dict) -> BaseArtifact:
+        profile_id = params["values"]["profile_id"]
+        header_dic = {"Authorization": "Bearer " + self.api_key}
+        params = {
+            "url": f"https://www.linkedin.com/in/{profile_id}",
+        }
+
+        response = requests.get(
+            self.api_endpoint, params=params, headers=header_dic, timeout=30
```
Comment (vasinov) [inline griptape/labs/tools/proxycurl/tool.py]: Let's move timeout into a property.

**Review by vasinov — CHANGES_REQUESTED** [review]
+tests

File: griptape/labs/tools/__init__.py [hunk 4]
```diff
@@ -7,6 +7,7 @@
 from .zoom.tool import Zoom
 from .openai_dalle.tool import OpenAiDallE
 from .sql_redshift.tool import SqlRedshiftClient
+from .proxycurl.tool import ProxycurlClient
```
Comment (vasinov) [inline griptape/labs/tools/__init__.py]: Let's call this package `proxycurl_client`, so it's consistent with other package names.

File: griptape/labs/tools/proxycurl/tool.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+@define
+class ProxycurlClient(BaseTool):
+    api_key: str = field(default=None, kw_only=True)
+
+    api_endpoint: str = field(default="https://nubela.co/proxycurl/api/v2/linkedin", kw_only=True)
+
+    @activity(
+        config={
+            "name": "get_profile",
```
Comment (vasinov) [inline griptape/labs/tools/proxycurl/tool.py]: Names in config are no longer required.

File: griptape/labs/tools/proxycurl/tool.py (hunk 1 above)
Comment (collindutter) [reply griptape/labs/tools/proxycurl/tool.py]: @vasinov I think `api_key` should really just be a required field. Don't think the requests will work without it.

**Pull request merged**